### ISSUE TITLE
[ENH] make sure that subject level GLM are run only on within brain data

### ISSUE THREAD

**Issue by Remi-Gau** [cpp-lln-lab/CPP_SPM#215]
Partly related to #164 

If a brain mask has been created from skullstripping we could be use directly as an implicit mask during the GLM:
- to avoid out of brain activations
- to make sure that the non-sphericity correction computation is only based on withing brain voxels.

- [x] A default option (when no skull-stripping was done for example) for MNI space is to use the ICV (intra cerebral volume) mask that comes with SPM.

**Comment by CerenB** [comment]
though... that brain mask is in anatomical space, the resolution is different right? that's why when I open it in mricron the overlap on top of the functional image is no good. 
Do you know what are the functions/code to have skullstripped anat brain mask in func dimensions? 

At the moment, I am using mean func data to skullstrip it and then save it as mask since it's in the "right" dimension. Pls see below:

```
% function uses a mean functional image to create individual space mask

    % STEP 1
    % go to mricron and create skull stripped mean functional image
    % by using FSL BET function
    % in the future think about implementing FSL BET into matlab
    [path, imageName, ext] = fileparts(imagePath);
    betImageName = ['bet05_', imageName, ext];
    betImagePath = fullfile(path, betImageName);

    % create mask name
    maskFileName = ['mask', betImageName];
    maskPath = fullfile(path, maskFileName);

    % STEP 2
    if ~exist(maskPath)
        % Create a template & load the mask
        % A = load_untouch_nii('bet_05_meanuasub-pil001-PitchFT_run-001.nii');
        A = load_untouch_nii(betImagePath);

        C = A;
        C.fileprefix = 'C';
        C.img = [];

        idx = find(A.img > 0);
        A.img(idx) = 1;
        C.img = A.img;
        save_untouch_nii(C, maskPath);
    end
```

**Comment by Remi-Gau** [comment]
thanks for the tip. 

yeah creating the mask for functional data is on the to do (see issue #164).

my idea was that since the functional should at least have been coregistered to anat, then we can simply reslice the brain mask obtained from the anat to the dimension of the functional. That should already do most of the job, no?

**Comment by CerenB** [comment]
Yup I totally agree with you! My only problem is I cannot find my previously done reslicing snippets...

**Comment by Remi-Gau** [comment]
partially solved for #443: will use SPM default ICV (intra cerebral volume) mask when running GLM in MNI space and no other mask has been defined.
